### ISSUE TITLE
Pyright treats dataclasses.dataclass(**kwargs) decorator calls as returning bool

### ISSUE THREAD

**Issue by FasterSpeeding** [microsoft/pyright#2102]
**Describe the bug**
Unwrapping a mapping to kwargs when calling `dataclasses.dataclass` when no other arguments are explicitly passed leads pyright treating the class it's wrapping as having been replaced by a bool.

**To Reproduce**
* Lint the example code
* Watch the reveal_type call be resolved to the following output
```
  ...\more_typing_tests.py:13:13 - info: Type of "Foo" is "bool"
```

**Expected behavior**
The following reveal_type call should resolve to `info: Type of "Bar" is "Type[Bar]"`. I'm not exactly sure how you'd want to handle deciding what dataclass features have been enabled in this case but it might be safe to either assume just the defaults or none I guess.

**Screenshots or Code**
```python
import dataclasses
import typing


KEYWORD_ARGS: typing.Dict[str, bool] = {}


@dataclasses.dataclass(**KEYWORD_ARGS)
class Foo:
    ...


reveal_type(Foo)

```

**VS Code extension or command-line**
pyright 1.1.157 (command-line tool)
targetting python 3.8

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

If you want `dataclass` to work with a static type checker, any parameters that affect the type (e.g. `final` or `init`) must be provided statically, not as a dynamic unpacked dictionary.

**Comment by erictraut** [comment]
This is included in pyright 1.1.158, which I just published. It will also be included in the next release of pylance.
